### PR TITLE
timg: init at v1.4.0

### PR DESCRIPTION
**Pull request by hzeller**
Timg is an image and video viewer for the terminal,
useful to look at images without leaving the comfort
of the shell or if remotely logged in to a shell.

https://timg.sh/

I am the maintainer of timg.

###### Motivation for this change

I want to use Nix on my remote webserver and have to assess images on that server; for that the timg tool is the ideal tool as it allows to present images in the terminal (Also, I am maintainer of timg so feel responsible to provide the package in NixOS).

###### Things done

(Note, I am very new to Nix and this is my first contribution to packages)

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [x] other Linux distributions  (`nix-build -A timg`)
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/graphics/timg/default.nix [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    "-DWITH_VIDEO_DEVICE=On"
+    "-DWITH_OPENSLIDE_SUPPORT=Off"  # https://openslide.org/ lib not yet in nix
+  ];
+
+  meta = with lib; {
+    homepage = "https://timg.sh/";
+    description = "A terminal image and video viewer";
+    license = licenses.gpl2;
+    platforms = platforms.unix;
+    maintainers = with maintainers; [ hzeller ];
```
Comment (hzeller) [inline pkgs/tools/graphics/timg/default.nix:50]: This is my first contribution. Do I have to enter myself in the `maintainers/maintainer-list.nix` list ?

Comment (hzeller) [reply pkgs/tools/graphics/timg/default.nix:50]: Looks like I have to. Added.

File: pkgs/tools/graphics/timg/default.nix [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+, graphicsmagick
+, libjpeg
+, ffmpeg
+, zlib
+, libexif
+}:
+
+stdenv.mkDerivation rec {
+  pname = "timg";
+  version = "v1.4.0";
```
Comment (OPNA2608) [inline pkgs/tools/graphics/timg/default.nix:16]:
```suggestion
  version = "1.4.0";
```

File: pkgs/tools/graphics/timg/default.nix [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+  src = fetchFromGitHub{
+    owner = "hzeller";
+    repo = "timg";
+    rev = version;
```
Comment (OPNA2608) [inline pkgs/tools/graphics/timg/default.nix:21]:
```suggestion
    rev = "v${version}";
```

File: pkgs/tools/graphics/timg/default.nix [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+  cmakeFlags = [
+    "-DWITH_VIDEO_DECODING=On"
+    "-DWITH_VIDEO_DEVICE=On"
+    "-DWITH_OPENSLIDE_SUPPORT=Off"  # https://openslide.org/ lib not yet in nix
+  ];
+
+  meta = with lib; {
+    homepage = "https://timg.sh/";
+    description = "A terminal image and video viewer";
+    license = licenses.gpl2;
```
Comment (OPNA2608) [inline pkgs/tools/graphics/timg/default.nix:48]: Please clarify if it's `gpl2Only` or `gpl2Plus`.

File: pkgs/tools/graphics/timg/default.nix (hunk 1 above)
Comment (OPNA2608) [reply pkgs/tools/graphics/timg/default.nix:50]: Yes, and this needs to be done in a separate commit before a package init that references you.

1. `maintainers: add hzeller`
2. `timg: init at 1.4.0`

File: pkgs/tools/graphics/timg/default.nix [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+    graphicsmagick
+    ffmpeg
+    libexif
+    libjpeg
+    zlib
+  ];
+
+  nativeBuildInputs = [
+    cmake
+    git
```
Comment (OPNA2608) [inline pkgs/tools/graphics/timg/default.nix:35]: What is git used for at build-time?

**Review by OPNA2608 — CHANGES_REQUESTED** [review]
Welcome to Nix! Made afew suggestions but it otherwise LGTM.

Result of `nixpkgs-review pr 119394` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>timg</li>
  </ul>
</details>

File: maintainers/maintainer-list.nix [hunk 6]
```diff
@@ -4045,6 +4045,12 @@
     githubId = 12491746;
     name = "Masato Yonekawa";
   };
+  hzeller = {
```
Comment (SuperSandro2000) [inline maintainers/maintainer-list.nix:4048]: Please put this into a separate commit with the message ``maintainers: add hzeller`` and sort it alphabetically.

Comment (hzeller) [reply maintainers/maintainer-list.nix:4048]: Done: #119428

File: pkgs/tools/graphics/timg/default.nix (hunk 2 above)
Comment (hzeller) [reply pkgs/tools/graphics/timg/default.nix]: Done.

File: pkgs/tools/graphics/timg/default.nix (hunk 3 above)
Comment (hzeller) [reply pkgs/tools/graphics/timg/default.nix]: Done.

File: pkgs/tools/graphics/timg/default.nix (hunk 4 above)
Comment (hzeller) [reply pkgs/tools/graphics/timg/default.nix]: Done (`gpl2Only`).

File: pkgs/tools/graphics/timg/default.nix (hunk 5 above)
Comment (hzeller) [reply pkgs/tools/graphics/timg/default.nix]: cmake rule attempts to extract version information if source is a git repository; it fails when it can't invoke git.

Added comment in default.nix for future reference.

Done.

Comment (hzeller) [reply pkgs/tools/graphics/timg/default.nix]: Fixed it in a better way now by telling `cmake` to not do that: ` -DTIMG_VERSION_FROM_GIT=Off`

File: maintainers/maintainer-list.nix (hunk 6 above)
Comment (Synthetica9) [reply maintainers/maintainer-list.nix:4048]: Seperate commit, not seperate PR

Comment (hzeller) [reply maintainers/maintainer-list.nix:4048]: Ah, yes, that I did actually as a separate commit: https://github.com/NixOS/nixpkgs/pull/119394/commits/569c9a1b767a51c3c72cf0f44ea95759b574ae7a
.. but I didn't know the commit message naming conventions back then, so the commit message is different. I am not sure if I can amend the message from a commit that is not at head, or can I ?

**Comment by SuperSandro2000** [comment]
Please squash all package related commits that we have two in the end.

**Comment by hzeller** [comment]
Done.
